arkfilepicker/src/main/java/space/taran/arkfilepicker/presentation/filepicker/ArkFilePickerFragment.kt

```diff
@@ -0,0 +1,429 @@
+                val outValue = TypedValue()
+                requireContext().theme.resolveAttribute(
+                    android.R.attr.selectableItemBackground,
+                    outValue,
+                    true
+                )
+                tv.setBackgroundResource(outValue.resourceId)
+                tv.setTextSize(TypedValue.COMPLEX_UNIT_SP, 20f)
+                tv.setPadding(padding, 0, 0, 0)
+                tv.isClickable = true
+                tv.text = text
+                if (pathWithoutDevice.last() == part)
+                    tv.setTextColor(
+                        resources.getColor(
+                            R.color.ark_file_picker_black,
+                            null
+                        )
+                    )
+                else {
+                    tv.setTextColor(
+                        resources.getColor(
+                            R.color.ark_file_picker_gray,
+                            null
+                        )
+                    )
+                    tv.setOnClickListener {
+                        viewModel.onItemClick(fullPathToPart)
+                    }
+                }
+                layoutPath.addView(tv)
+            }
+        scrollPath.post {
+            scrollPath.fullScroll(ScrollView.FOCUS_RIGHT)
+        }
+    }
+
+    override fun getTheme() = themeId!!
+
+    private fun initBackButtonListener() {
+        requireDialog().setOnKeyListener { _, keyCode, keyEvent ->
+            if (keyCode == KeyEvent.KEYCODE_BACK &&
+                keyEvent.action == KeyEvent.ACTION_UP
+            ) {
+                viewModel.onBackClick()
+            }
+            return@setOnKeyListener true
+        }
+    }
+
+    companion object {
+        const val PATH_PICKED_REQUEST_KEY = "arkFilePickerPathPicked"
+        const val PATH_PICKED_PATH_BUNDLE_KEY = "arkFilePickerPathPickedPathKey"
+        const val FOLDER_CHANGED_REQUEST_KEY = "arkFilePickerFolderChanged"
+        const val FOLDER_CHANGED_FOLDER_BUNDLE_KEY =
+            "arkFilePickerFolderChangedFolderKey"
+
+        fun newInstance(config: ArkFilePickerConfig) =
+            ArkFilePickerFragment().apply {
+                titleStringId = config.titleStringId
+                pickButtonStringId = config.pickButtonStringId
+                cancelButtonStringId = config.cancelButtonStringId
+                internalStorageStringId = config.internalStorageStringId
+                accessDeniedStringId = config.accessDeniedStringId
+                itemsPluralId = config.itemsPluralId
+                themeId = config.themeId
+                initialPath = config.initialPath?.toString()
+                showRoots = config.showRoots
+                pathPickedRequestKey = config.pathPickedRequestKey
+                mode = config.mode.ordinal
+            }
+
+        private const val DIALOG_WIDTH = 300f
+        private const val PATH_PART_PADDING = 4f
+    }
+}
+
+internal class FilesPage(
```

Comment:
Maybe "FilesTab"?

Reply:
For me a tab is more like a title/button for navigating to a page. I think the `FilesPage` is the correct name since it is used in the `ViewPager`

Reply:
Ok if `ViewPager` is conventional name